servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/ExceptionUtil.java

```diff
@@ -16,186 +16,118 @@
             return new MessageNotFoundException(errorCondition.getDescription());
-        }
-        else if (errorCondition.getCondition() == ClientConstants.ENTITY_ALREADY_EXISTS_ERROR)
-        {
+        } else if (errorCondition.getCondition() == ClientConstants.ENTITY_ALREADY_EXISTS_ERROR) {
             return new MessagingEntityAlreadyExistsException(errorCondition.getDescription());
-        }
-        else if (errorCondition.getCondition() == AmqpErrorCode.DecodeError)
-        {
+        } else if (errorCondition.getCondition() == AmqpErrorCode.DecodeError) {
             return new ServiceBusException(false, new AmqpException(errorCondition));
         }
 
         return new ServiceBusException(ClientConstants.DEFAULT_IS_TRANSIENT, errorCondition.toString());
     }
 
-    static <T> void completeExceptionally(CompletableFuture<T> future, Throwable exception, IErrorContextProvider contextProvider, boolean completeAsynchronously)
-    {
-        if (exception != null && exception instanceof ServiceBusException)
-        {
+    static <T> void completeExceptionally(CompletableFuture<T> future, Throwable exception, IErrorContextProvider contextProvider, boolean completeAsynchronously) {
+        if (exception != null && exception instanceof ServiceBusException) {
             ErrorContext errorContext = contextProvider.getContext();
             ((ServiceBusException) exception).setContext(errorContext);
         }
 
-        if(completeAsynchronously)
-        {
+        if(completeAsynchronously) {
             AsyncUtil.completeFutureExceptionally(future, exception);
-        }
-        else
-        {
+        } else {
             future.completeExceptionally(exception);
         }
     }
 
     // not a specific message related error
-    static boolean isGeneralError(Symbol amqpError)
-    {
+    static boolean isGeneralError(Symbol amqpError) {
         return (amqpError == ClientConstants.SERVER_BUSY_ERROR
                 || amqpError == ClientConstants.TIMEOUT_ERROR
                 || amqpError == AmqpErrorCode.ResourceLimitExceeded);
     }
 
-    static String getTrackingIDAndTimeToLog()
-    {
+    static String getTrackingIDAndTimeToLog() {
         return String.format(Locale.US, "TrackingId: %s, at: %s", UUID.randomUUID().toString(), ZonedDateTime.now());
     }
 
-    static String toStackTraceString(final Throwable exception, final String customErrorMessage)
-    {
+    static String toStackTraceString(final Throwable exception, final String customErrorMessage) {
         final StringBuilder builder = new StringBuilder();
 
-        if (!StringUtil.isNullOrEmpty(customErrorMessage))
-        {
+        if (!StringUtil.isNullOrEmpty(customErrorMessage)) {
             builder.append(customErrorMessage);
             builder.append(System.lineSeparator());
         }
 
         builder.append(exception.getMessage());
-        if (exception.getStackTrace() != null)
```

Comment:
[SpotBugs-P3]
Redundant nullcheck of Throwable.getStackTrace(), which is known to be non-null in com.microsoft.azure.servicebus.primitives.ExceptionUtil.toStackTraceString(Throwable, String)